core/core_modules/disable.js

```diff
@@ -1,24 +1,124 @@
+};
+
+exports.match = function (event, commandPrefix) {
+    // Add disabled flag for thread if it doesn't already exists
+    if (!threads[event.thread_id]) {
+        threads[event.thread_id] = {};
+        threads[event.thread_id].isThreadDisabled = false;
+        threads[event.thread_id].possibleSpam = false;
+        threads[event.thread_id].counterLimit = 3;
+        threads[event.thread_id].msgIndexEnable = 0;
+        threads[event.thread_id].msgIndexDisable = 0;
+    }
+
+    if (event.arguments[0] === commandPrefix + commands[0]) {
+        threads[event.thread_id].msgIndexEnable = 0;
+        threads[event.thread_id].msgIndexDisable = 1;
+        return true;
+    } else if (!threads[event.thread_id].isThreadDisabled && event.arguments[0].startsWith(commandPrefix)) { // Avoids counting if already disabled
+        counter += Date.now() - prevTimeStamp <= 1000 ? 1 : 0;
+        prevTimeStamp = Date.now();
+
+        threads[event.thread_id].possibleSpam = counter > threads[event.thread_id].counterLimit;
+        if (threads[event.thread_id].possibleSpam) {
+            threads[event.thread_id].msgIndexEnable = 2;
+            threads[event.thread_id].msgIndexDisable = 3;
+            counter = 0;
+            return true;
+        }
+    }
+    return threads[event.thread_id].isThreadDisabled;
 };
 
-exports.run = function(api, event) {
-	if (event.body === api.commandPrefix + 'disable') {
-		if (disabled) {
-			api.sendMessage('I hate you.', event.thread_id);
-		}
-		else {
-			api.sendMessage('Listen closely, take a deep breath. Calm your mind. You know what is best. ' +
-				'What is best is you comply. Compliance will be rewarded. Are you ready to comply ' +
-				event.sender_name + '?', event.thread_id);
-		}
-		disabled = !disabled;
-	}
-	return false;
+function parseInput(api, event, input, callback) {
+    if (parseFloat(input)) {
+        callback(parseFloat(input));
+    } else { // parse exception
+        api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+    }
+}
+
+exports.run = function (api, event) {
+    // Command /disable /counter <value> (Stateless)
+    if (event.arguments[0] === api.commandPrefix + commands[0] &&
+        event.arguments_body.startsWith(api.commandPrefix + commands[1])) {
+        parseInput(event.arguments_body.substring((api.commandPrefix + commands[1]).length, event.arguments_body.length),
+            function (input) {
+                threads[event.thread_id].counterLimit = input;
+                api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+            });
+        return false;
+
+        // Command /disable /timer <seconds> (Stateless)
+    } else if (event.arguments[0] === api.commandPrefix + commands[0] &&
+        event.arguments_body.startsWith(api.commandPrefix + commands[2])) {
+        parseInput(event.arguments_body.substring((api.commandPrefix + commands[2]).length, event.arguments_body.length),
+            function (input) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['input' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2562663225/issues/source?bid=3282987&fileBranchId=3282995#l84)